internal/api/middleware.go

```diff
@@ -152,6 +147,67 @@ func authMiddleware(ctx *middlewareContext) middleware {
 	}
 }
 
+func getClaimsFromAuthorizationHeader(header string, jwtSecret []byte) (*jwtGocertClaims, error) {
+	if header == "" {
+		return nil, fmt.Errorf("authorization header not found")
+	}
+	bearerToken := strings.Split(header, " ")
+	if len(bearerToken) != 2 || bearerToken[0] != "Bearer" {
+		return nil, fmt.Errorf("authorization header couldn't be processed. The expected format is 'Bearer <token>'")
+	}
+	claims, err := getClaimsFromJWT(bearerToken[1], jwtSecret)
+	if err != nil {
+		return nil, fmt.Errorf("token is not valid: %s", err)
+	}
+	return claims, nil
+}
+
+// AllowRequest looks at the user data to determine the following things:
+// The first question is "Is this user trying to access a path that's restricted?"
+//
+// There are two types of restricted paths: admin only paths that only admins can access, and self authorized paths,
+// which users are allowed to use only if they are taking an action on their own user ID. The second question is
+// "If the path requires an ID, is the user attempting to access their own ID?"
+//
+// For all endpoints and permission permutations, there are only 2 cases when users are allowed to use endpoints:
+// If the URL path is not restricted to admins
+// If the URL path is restricted to self authorized endpoints, and the user is taking action with their own ID
+// This function validates that the user the with the given claims is allowed to use the endpoints by passing the above checks.
+func AllowRequest(claims *jwtGocertClaims, method, path string, permissionRestrictions []struct {
```

Comment:
RestrictedPaths is only used inside of this func right? Why not have it inside of the func instead of taking it as a parameter. Then the inputs can only be claims, path and the returns can only be bool, error.

Reply:
yep. done